### PR TITLE
Added NNBuilder utility with Torch Sequential builder

### PR DESCRIPTION
**Pull request by cabesd**

### CODE REVIEW

File: torchagents/utilities/NNBuilder.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import torch
+from typing import List
+
+def LinearSequentialBuilder(
+        input_size: int,
+        output_size: int,
+        hidden_layers: List[int],
+        output_function: torch.nn = None,
+):
```
Comment (michaelmunje) [inline torchagents/utilities/NNBuilder.py:9]: Is there a reason why "):" is in a newline?

File: test/test_NNBuilder.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def tearDown(self):
+        pass
+
+    def setUp(self):
+        self.config = {'input_size': np.random.randint(10), 'output_size': np.random.randint(10), 'hidden_layers': [128, 128, 256], 'output_function': torch.nn.Softmax}
+        self.nn = NNBuilder.LinearSequentialBuilder(**self.config)
+
+    def test_linear_sequential_builder_output(self):
+        t = torch.rand(self.config['input_size'])
+        assert self.nn(t).shape == torch.Size([self.config['output_size']])
```
Comment (michaelmunje) [inline test/test_NNBuilder.py:17]: Note that you could also do self.assertTrue(a, b).

This might be better for consistency but is minor.

File: test/test_NNBuilder.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import numpy as np
+from torchagents.utilities import NNBuilder
+
+class TestNNBuilder(TestCase):
+
+    def tearDown(self):
+        pass
+
+    def setUp(self):
+        self.config = {'input_size': np.random.randint(10), 'output_size': np.random.randint(10), 'hidden_layers': [128, 128, 256], 'output_function': torch.nn.Softmax}
```
Comment (michaelmunje) [inline test/test_NNBuilder.py:12]: I think the random seed should be fixed here.

File: torchagents/utilities/NNBuilder.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    activation_function = torch.nn.ReLU
+    net_layers = []
+
+    # Append input layers
+    first_hidden_layer_size = hidden_layers.pop(0)
```
Comment (michaelmunje) [inline torchagents/utilities/NNBuilder.py:14]: Making net_layers looks like it could be a single line list comprehension.

Comment (michaelmunje) [reply torchagents/utilities/NNBuilder.py:14]: Hmm, looking again since we need to keep track of the pairs

Comment (michaelmunje) [reply torchagents/utilities/NNBuilder.py:14]: Yes, it should be feasible to turn this into a list comprehension. This won't make a different in this small test case but for others it should be faster since append is an O(n) operation.

Comment (cabesd) [reply torchagents/utilities/NNBuilder.py:14]: Sure. That might require having to add @input_size and @output_size to the @layers list.
Although, take into account that this will ONLY add overhead and not affect the RL loop runtime.

File: test/test_NNBuilder.py (hunk 3 above)
Comment (cabesd) [reply test/test_NNBuilder.py:12]: Agree!

File: test/test_NNBuilder.py (hunk 2 above)
Comment (cabesd) [reply test/test_NNBuilder.py:17]: Got it

File: torchagents/utilities/NNBuilder.py (hunk 1 above)
Comment (cabesd) [reply torchagents/utilities/NNBuilder.py]: No reason
